File: springboot_CRUD_board/src/main/java/com/study/board/service/BoardService.java
package com.study.board.service;

import com.study.board.entity.Board;
import com.study.board.repository.BoardRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;

import java.io.File;
import java.io.IOException;
import java.util.List;
import java.util.UUID;

@Service
public class BoardService {
    @Autowired
    private BoardRepository boardRepository;
    public void write(Board board, MultipartFile file) throws Exception{
        String projectPath = System.getProperty("user.dir") + "\\src\\main\\resources\\static\\img";
        UUID uuid = UUID.randomUUID();
        String fileName = uuid + "_" + file.getOriginalFilename();
        File saveFile = new File(projectPath, fileName);
        file.transferTo(saveFile);
        board.setFilename(fileName);
        board.setFilepath("/img/" + fileName);
        boardRepository.save(board);
    }
    public Page<Board> boardList(Pageable pageable) {
        return boardRepository.findAll(pageable);
    }
    public Page<Board> boardSearchList(String searchKeyword, Pageable pageable) {
        return boardRepository.findByTitleContaining(searchKeyword, pageable);
    }
    public Board boardView(Integer id) {
        return boardRepository.findById(id).get();
    }
}

File: springboot_CRUD_board/src/main/kotlin/comirotod/service/BoardService.kt
package comirotod.service

import comirotod.entity.Board
import comirotod.repository.BoardRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service
import org.springframework.web.multipart.MultipartFile
import java.io.File
import java.util.*

@Service
class BoardService @Autowired constructor(
    private val boardRepository: BoardRepository
) {
    @Throws(Exception::class)
    fun write(board: Board, file: MultipartFile) {
        val projectPath = System.getProperty("user.dir") + "\\src\\main\\resources\\static\\img"
        val uuid = UUID.randomUUID()
        val fileName = uuid.toString() + "_" + file.originalFilename
        val saveFile = File(projectPath, fileName)
        file.transferTo(saveFile)
        board.filename = fileName
        board.filepath = "/img/$fileName"
        boardRepository.save(board)
    }
    fun boardList(pageable: Pageable?): Page<Board?> {
        return boardRepository.findAll(pageable!!)
    }
    fun boardSearchList(searchKeyword: String?, pageable: Pageable?): Page<Board?>? {
        return boardRepository.findByTitleContaining(searchKeyword!!, pageable!!)
    }
    fun boardView(id: Int): Board? {
        return boardRepository.findById(id).orElseThrow { RuntimeException("Board not found") }
    }
}